x-pack/plugins/security_solution/public/resolver/view/use_resolver_query_params.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useMemo } from 'react';
+// eslint-disable-next-line import/no-nodejs-modules
+import querystring from 'querystring';
+import { useHistory, useLocation } from 'react-router-dom';
+import { CrumbInfo } from './panels/panel_content_utilities';
+
+export function useResolverQueryParams(documentLocation: string) {
```

Comment:
:grey_question: My only question here would be if this is "passable" in a URL they way we envisioned. I think we put it in this way now regardless, but maybe we should test that someone on the same Kibana instance can always open the URL and restore/rehydrate context based on this.